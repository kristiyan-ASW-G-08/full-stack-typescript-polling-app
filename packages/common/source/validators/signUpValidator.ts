import * as yup from 'yup';

const signUpValidator = yup.object().shape({
  username: yup
    .string()
    .trim()
    .min(1)
    .max(50)
    .required(),
  location: yup
    .string()
    .trim()
    .min(1)
    .required(),
  email: yup
    .string()
    .trim()
    .email()
    .required(),
  password: yup
    .string()
    .trim()
    .min(12)
    .required(),
  confirmationPassword: yup
    .string()
    .trim()
    .min(12)
    .oneOf([yup.ref('password')], 'Passwords must match')
    .required(),
});

export default signUpValidator;
