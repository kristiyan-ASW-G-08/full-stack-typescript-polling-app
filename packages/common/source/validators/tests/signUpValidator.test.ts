import signUpValidator from '@customValidators/loginValidator';

describe('loginValidators', (): void => {
  const validValidationTargets = [
    {
      username: 'TestUser',
      email: 'validTest@test.test',
      password: 'validPassword',
      confirmationPassword: 'validPassword',
      location: 'London',
    },
    {
      username: 'SecondUser',
      email: 'testTest@test.test',
      password: 'pass1012asd1212',
      confirmationPassword: 'pass1012asd1212',
      location: 'Sofia',
    },
  ];
  const invalidValidationTargets = [
    {
      email: 'validTest@test',
      password: 'validPassword',
      confirmationPassword: 'pass1012asd1212',
      location: 'Sofia',
      username: 'testUser',
    },
    {
      email: 'testTest.test',
      password: '1234',
      confirmationPassword: 'pass1012asd1212',
      location: 'Sofia',
      username: '',
    },
    {},
    {
      unknown: 'testTest.test',
      password: '1234',
      confirmationPassword: 'pass1012asd1212',
      location: 'Sofia',
      username: '',
    },
  ];
  it.each(validValidationTargets)(
    'validate should resolve successfully',
    async target => {
      await expect(
        signUpValidator.validate(target, { abortEarly: false }),
      ).resolves.toStrictEqual(target);
    },
  );
  it.each(invalidValidationTargets)('validate should reject', async target => {
    await expect(
      signUpValidator.validate(target, { abortEarly: false }),
    ).rejects.toThrowErrorMatchingSnapshot();
  });
});
