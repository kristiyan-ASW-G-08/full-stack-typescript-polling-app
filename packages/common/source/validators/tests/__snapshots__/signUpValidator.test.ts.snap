// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loginValidators validate should reject 1`] = `"email must be a valid email"`;

exports[`loginValidators validate should reject 2`] = `"2 errors occurred"`;

exports[`loginValidators validate should reject 3`] = `"2 errors occurred"`;

exports[`loginValidators validate should reject 4`] = `"2 errors occurred"`;
